plugins/net_plugin/net_plugin.cpp

```diff
@@ -2224,6 +2224,8 @@ namespace eosio {
          set_state( lib_catchup );
          sync_last_requested_num = 0;
          sync_next_expected_num = chain_info.lib_num + 1;
+      } else if (sync_next_expected_num >= sync_last_requested_num) {
+         // break
```

Comment:
This change causes the net_plugin to request more blocks if possible instead of remaining in a mode where it thinks it is already syncing.